add HIL_STATE message handler and publish to uORB

kd0aij opened the pull request:
@bkueng Incorporates https://github.com/PX4/Firmware/pull/5323

add missing break

uorb topics generator: add multi-topics to the list of all topics

topic names with '# TOPICS <name>' were previously not in orb_get_topics().
This means the logger could not find them.

Affects for example actuator_controls_0.

px_generate_uorb_topic_files.py: add multitopics to generate_topics_list_file_from_files method

switch simulated attitude to new topic: vehicle_attitude_groundtruth

logger: add input_rc topic. needed for web plotting

input_rc.msg: remove timestamp_publication, use timestamp instead

mixer.cpp: warnx -> PX4_ERR

logger: initialize timer_call to 0 (hrt_call_every reads some fields)

position_setpoint_triplet topic: set the timestamp when publishing

px_generate_uorb_topic_files.py: add multitopics to generate_topics_list_file_from_files method

add vehicle_attitude_groundtruth to default topics

change to hil_state_quaternion

bkueng commented on src/modules/logger/logger.cpp:
```diff
@@ -476,6 +476,7 @@ bool Logger::copy_if_updated_multi(LoggerSubscription &sub, int multi_instance,
 void Logger::add_default_topics()
 {
 	add_topic("vehicle_attitude", 10);
+	add_topic("vehicle_attitude_groundtruth", 10);
```
Should we add an ifdef here for SITL? Because the logger needs to iterate all subscribed topics on each interval, and checking for a topic that is never published just wastes CPU cycles.

kd0aij replied:
I think that would be better than having to use a custom topic list on the SD card.

bkueng commented on src/modules/simulator/simulator_mavlink.cpp:
```diff
@@ -301,7 +303,40 @@ void Simulator::handle_message(mavlink_message_t *msg, bool publish)
+			hil_attitude.pitch = euler(0);
+			hil_attitude.yaw = euler(2);
+
+			if (true) {
```
What's the intention with this?

kd0aij replied:
I forgot about that :) the "publish" argument was false, and I didn't know where it was set. Regarding the order of Euler angles, it seems that jMAVSim has roll and pitch swapped, but that's mind-boggling, or at least left-handed.

bkueng commented:
@kd0aij we also need to add `logger start -e -t` to the config, right? I think we can even run both logger's for SITL (I already do that without problems).

kd0aij commented:
Yes, I changed my own configs to run logger instead of sdlog2, and forgot to pull that over here. Probably better not to run both since that would just be a waste of electricity.

kd0aij commented:
@bkueng I figured out how the "publish" arg is intended to work (I think) and fixed that. I also changed the  config for posix_sitl_default to run the new logger instead of sdlog2.
Just need to straighten out the issues with PX4/jMAVSim and jMAVlib now.

bkueng commented:
Ok we can use the logger only, but I'll forward any complaints as to why there are no .px4logs anymore to you :)

I don't think the -p option is what we want, because:
- this adds another gyro/baro/accel publisher, but they are already published in accelsim (the same data)
- I get `ERROR [sensors] Baro #0 failover : Stale data!`

So you can just unconditionally publish ground-truth, since it's kind of a special topic.

kd0aij commented:
Guess I can gripe some more about a lack of documentation since I guessed wrong about what the "-p" option was meant for. Also stupidity on my part for not realizing this would stomp on the existing messages instead of creating new instances. But perhaps new instances were actually the intended result?

 I'll just delete the if statement and go back to "-s".

kd0aij commented:
@bkueng Let me know if there's anything preventing this from being merged and I'll try to fix it.

bkueng commented:
No it's fine. I rebased & applied it
